PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/util/SwingExts.kt

```diff
@@ -24,15 +24,17 @@ package com.microsoft.intellij.ui.util
 
 import javax.swing.ListModel
 
-val <T> ListModel<T>.iterator: Iterator<T>
-    get() = this.let {
+val <T> ListModel<T>?.iterator: Iterator<T>
+    get() = if (this != null) {
```

Comment:
How about writing in the following way?
```
val <T> ListModel<T>?.iterator: Iterator<T>
    get() = this?.let {
        object: Iterator<T> {
            private var position: Int = 0

            override fun hasNext() = size > position
            override fun next() = getElementAt(position++)
        }
    } ?: emptyList<T>().iterator()

```

Reply:
I've tried this before, there must be a compiler or class loader issue here:
```
java.lang.NoClassDefFoundError: com/microsoft/intellij/ui/util/SwingExtsKt$iterator$1$1

	at com.microsoft.intellij.ui.util.SwingExtsKt.getIterator(SwingExts.kt:28)
	at com.microsoft.intellij.ui.util.SwingExtsKt.findFirst(SwingExts.kt:37)
```